seisbench/generate/augmentation.py

```diff
@@ -670,3 +672,194 @@ def __call__(self, state_dict):
 
     def __str__(self):
         return f"GaussianNoise (Scale (mu={self.scale[0]}, sigma={self.scale[1]}))"
+
+
+class RotateHorizontalComponents:
+    """
+    Rotates horizontal components either by a given value alpha in rad or by an arbitrary angle between -pi and +pi.
+
+    :param alpha: Angle in rad for rotation of horizontal components.
+                  If alpha is zero, a new angle is defined each time the method is called, otherwise the angle is
+                  constant.
+                  Default value is None, i.e. the angle changes with every call.
+    :type alpha: None, float, int
+    :param components: Defines which components are rotated.
+                       Default components are N and E.
+    :type components: str
+    :param key: The keys for reading from and writing to the state dict.
+                If key is a single string, the corresponding entry in state dict is modified.
+                Otherwise, a 2-tuple is expected, with the first string indicating the key to
+                read from and the second one the key to write to.
+    :type key: str, tuple[str, str]
+    """
+
+    def __init__(
+        self, key: str = "X", alpha: (None, float, int) = None, components: str = "NE"
+    ):
+        if isinstance(key, str):
+            self.key = (key, key)
+        else:
+            self.key = key
+
+        self.components = components
+        self.alpha = alpha
+
+        if len(self.components) != 2:
+            msg = f"Only two components can be rotated and not {len(self.components)}."
+            raise ValueError(msg)
+
+    def __call__(self, state_dict) -> np.array:
+        x, metadata = state_dict[self.key[0]]
+
+        if self.key[0] != self.key[1]:
+            # Ensure metadata is not modified inplace unless input and output key are anyhow identical
+            metadata = copy.deepcopy(metadata)
+
+        # Create random angle alpha if alpha is not given in init
+        if self.alpha is None:
+            alpha = np.random.uniform(-np.pi, np.pi)
+        else:
+            alpha = self.alpha
+
+        trace_components = metadata.get("trace_component_order")
+        if not trace_components:
+            msg = "Keyword 'trace_component_order' is missing in metadata, therefore rotation is not possible."
+            raise ValueError(msg)
+
+        data_comp1 = copy.copy(x[trace_components.index(self.components[0]), :])
```

Comment:
I think the case that the component is not in the `trace_components` should be caught explicitly to give a more descriptive error of what's gone wrong.